README.md

```diff
@@ -24,10 +24,65 @@ A sample implementation of this can be found in the file `sample-main.go`
 and `pkg/sample-cmd` directory.  You'll want to have the equivalent files
 in your project.
 
-### A note on Dependencies
+### Example for developing your own Custom Metric Server API
+This is an example of how to vendor this project and setup your own provider.  This example uses [glide](https://github.com/Masterminds/glide) but you can use any other [dependency management tool](https://github.com/golang/go/wiki/PackageManagementTools) of your choice.
 
-You'll need to `glide install` dependencies before you can use this
-project.
+#### Create your new project:
+
+- `mkdir $GOPATH/src/github.com/your-company/my-custom-metric-server`
+- `cd $GOPATH/src/github.com/your-company/my-custom-metric-server`
+
+#### Vendor this repository:
+
+- `glide create`
+- `glide get github.com/kubernetes-incubator/custom-metrics-apiserver`
+
+#### Create the entry point from the sample:
+
+- `touch main.go` 
+
+> See the [sample entry point](https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/master/sample-main.go) for a complete example
+
+#### Add the server:
+
+- `mkdir -p pkg/cmd/server`
+- `touch pkg/cmd/server/start.go` 
+
+> See the [sample server](https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/master/pkg/sample-cmd/server/start.go) for complete example
+
+#### Add and implement [the custom metrics interface](https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/d8f23423aa1d0ff2bc9656da863d721725b3c68a/pkg/provider/interfaces.go#L84) for your provider:
+
+- `mkdir -p pkg/cmd/provider`
+- `touch pkg/cmd/server/provider.go` 
+
+> See the [sample provider](https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/master/pkg/sample-cmd/provider/provider.go) for complete example
+
+#### Build the project
+After adding and implementing the files above build you project and run it.
+
+## Development for boilerplate project
+
+### Pre-reqs
+
+- [glide](https://github.com/Masterminds/glide#install) to install dependencies before you can use this project.
+- [Go](https://golang.org/doc/install) 1.8+ 
```

Comment:
1.10+ at this point.  Might just want to say "same version of Go as kubernetes" and provide a link, so that we don't have to change it every time kubernetes does (otherwise, in my experience, it gets out of date).